src/blockdev.rs

```diff
@@ -388,8 +387,16 @@ impl Partition {
         // Now assume a kpartx "partition", where the path is a symlink to
         // an unpartitioned DM device node.
         // /sys/block/dm-1
-        let (target, is_link) = resolve_link(&self.path)?;
+        let target = PathBuf::from(&self.path);
+        let is_link = target
+            .symlink_metadata()
+            .chain_err(|| format!("reading metadata for {}", target.display()))?
+            .file_type()
+            .is_symlink();
         if is_link {
+            let target = target
```

Comment:
The initial value of `target` is misleading, I think.  What about this?

```suggestion
        if symlink_metadata(&self.path)
            .chain_err(|| format!("reading metadata for {}", &self.path))?
            .file_type()
            .is_symlink() {
            let target = Path::new(&self.path)
```